core/src/connection_reuse.rs

```diff
@@ -162,66 +151,44 @@ where
         Ok((Box::new(listener) as Box<_>, new_addr))
     }
 
-    fn dial(self, addr: Multiaddr) -> Result<Self::Dial, (Self, Multiaddr)> {
+    fn dial(&self, addr: Multiaddr) -> TransportResult<Self::Dial> {
+        // FIXME: this doesn't properly work yet.
         // If we already have an active connection, use it!
-        let substream = if let Some(muxer) = self.shared
-            .lock()
-            .active_connections
-            .get(&addr)
-            .map(|muxer| muxer.clone())
-        {
-            let a = addr.clone();
-            Either::A(muxer.outbound().map(move |s| s.map(move |s| (s, future::ok(a)))))
-        } else {
-            Either::B(future::ok(None))
-        };
+        // if let Some(muxer) = self.shared.lock().active_connections.get(&addr).map(|m| m.clone()) {
+        //     let a = addr.clone();
+        //     let f = muxer.outbound()
+        //         .map(move |s| s.map(move |s| (s, future::ok(a))))
+        //         .map(|o| Either::A(future::ok(o)));
+        //     let b = Box::new(f);
+        //     return Ok(b)
+        // }
```

Comment:
Would this not disable connection re-use for all dials?

Reply:
Right. In my efforts to put forward the general approach, and knowing that tomaka is working on re-doing this entire part, I temporarily disabled this part as it wasn't easy to restructure. I had forgotten that I still needed to convert this. It's done now.